### ISSUE TITLE
Support aarch64 Linux with page sizes >= 16k

### ISSUE THREAD

**Issue by maximbaz** [swc-project/swc#5967]
### Describe the bug

swc uses jemalloc, which in turn uses the build system's page size as internal page size, and fails if the target's page size is larger [1]. Some Linux distributions patch jemalloc itself (e.g. [2]), but Github actions uses a "standard" jemalloc, which has this problem.

Instead of using a patched jemalloc, it is possible to configure jemalloc in runtime to use largest possible size, thus making it compatible with all `aarch64` systems.

**The solution is simple: add `export JEMALLOC_SYS_WITH_LG_PAGE=16` before running `cargo build` for `aarch64` system.**

This is a very common case with rust projects, as you can imagine 🙂 Here's just one example [3].

Let me know if it makes sense!

1: https://github.com/jemalloc/jemalloc/issues/467
2: https://github.com/archlinuxarm/PKGBUILDs/blob/master/extra/jemalloc/PKGBUILD
3: https://github.com/archlinuxarm/PKGBUILDs/pull/1940

### Expected behavior

No crash on Linux with 16k page size.

### Version

1.3.3

**Comment by kdy1** [comment]
@Brooooooklyn Thoughts?
Do you think this worths the effort?

**Comment by mstephenson6** [comment]
Ran into this on Asahi (Arch) Linux on Apple M1 today trying to `npm install @swc/core`. In a fresh working directory with no `node_modules`, the install seems to finish successfully. But when installing `@swc/core` in a long list of `package.json` dependencies, the postinstall step fails with:

```
<jemalloc>: Unsupported system page size
memory allocation of 160 bytes failed
```

I traced this to `binding.js`, when it imports `@swc/core-linux-arm64-gnu/swc.linux-arm64-gnu.node` along the same lines as this issue discusses. The `swc` executable in that directory will run on my M1 Linux system, however, which I cannot explain.
